lib/src/http/http_provider.dart

```diff
@@ -210,12 +207,19 @@ class HttpProvider extends Provider with FluriMixin {
     if (reqUri == null || reqUri.toString() == '') throw new StateError(
         'HttpProvider: Cannot send a request without a URI.');
 
+    // If this is a retry, there will be a previous request stored in the meta.
+    HttpContext previousAttempt =
+        meta.containsKey('previousAttempt') ? meta['previousAttempt'] : null;
```

Comment:
it looks like you're trying to remove `meta` from use by all 'normal' logic in w_service.  If you're removing the use of `meta` in all of the retry logic, any reason to keep using it for `previousAttempt`?